### DISCUSSION TITLE
How can I add an API key to my .env file?

### DISCUSSION THREAD

**Discussion by FrankDaze** [mudler/LocalAI#2616] [General]
Hi, I like to protect my API with an API key. But I have no idea about the env variable name.
I tried API_KEY and LOCALAI_API_KEY, but nothing has any effect. 

Can someone tell me the correct var name?

Thanks in Advance
Frank

**Comment by jtwolfe** [comment]
you were correct it is 'API_KEY'
see the[ QuickStart documentation](https://localai.io/basics/getting_started/)

**Reply by JOduMonT** [reply]
> you were correct it is 'API_KEY'
> see the[ QuickStart documentation](https://localai.io/basics/getting_started/)

I would like to understand how to add an API_KEY in my .env (example)
I generated a random string with `openssl rand -base64 42` as I usually do with DiFy and Open-WebUI
but can't find where to add it in .env for LocalAI.

Also; for now I protect my Public LocalAI instance with Cloudflare ZeroTrust; and don't want to use Requestly

#### so if someone could explain me ...
## WHY 
why it is not in .env nor https://localai.io/advanced/#environment-variables
and when I go deep down digging in https://github.com/mudler/LocalAI/issues/2227#issuecomment-2106383136
and add LOCALAI_API_KEY="MY GENERATED API" in .env it still not working properly.

**Comment by FrankDaze** [comment]
thanks I think my problem was related to an older version of the image.

**Comment by JOduMonT** [comment]
> Can someone tell me the correct var name?
> 
> Thanks in Advance

https://github.com/mudler/LocalAI/issues/2227#issuecomment-2466143580

I use Docker Compose
in my .env I added the variable

> LOCALAI_API_KEY="GENERATED KEY"

I generated my key with: `openssl rand -base64 42`
